### ISSUE TITLE
[Bug]: SimpleX keeps server communication on the background

### ISSUE THREAD

**Issue by Yobo7000** [simplex-chat/simplex-chat#4213]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Platform

Android

### OS version

LineageOS 20

### App version

5.7.*

### Current Behavior

SimpleX keeps communication with SMP servers while notifications disabled and battery usage "restricted"

### Expected Behavior

imo this is nasty behavior and I'd like to disable any network communications while the app is closed

### Steps To Reproduce

1) set notifications to "allow only when the app is open"
2) disable app's battery usage in the app settings
3) monitor network traffic using "pcapdroid" or something similar
-> you will see that SimpleX is doing that nasty things regardless of all options are set to stop it

### Relevant log output

```shell
no log for this issue
```

**Comment by Yobo7000** [comment]
> how do you close the app, do you fully shutdown it within Android shutdown apps menu or do you just close its window?

I just close it and it kinda keeps running on the background but imo this is still unwanted behavior

**Comment by Yobo7000** [comment]
> that is expected Android behavior, it does not shutdown apps after close, if they have any background process
all other apps I have doesn't communicate on the background or at least they have option to disable it. SimpleX is the only doing it so as far as I understand it is misfunction

**Comment by Yobo7000** [comment]
> I am not, just saying that this is expected behavior in Android devices and it would be non-intuitive for most users use-cases to make the app non-functional whenever it goes out of focus, yes it could be an option, but that's unlikely to have many uses...

fella you made just 2 commits in this project and you are already brave enough to speak like you are a developer and know something however I am pretty sure you just started to learn how to write function in haskell but you still have no idea about proper software design. 

please keep you ignorance beyond the keyboard and don't comment this this topic anymore. I started this topic for people who know what they are doing.

**Comment by nahuhh** [comment]
...

1. menu > Notifications > run when app is open
2. Menu > shutdown

**Comment by Yobo7000** [comment]
>     1. menu > Notifications > run when app is open
>     2. Menu > shutdown

that option doesn't have any effect

**Comment by nahuhh** [comment]
> >     1. menu > Notifications > run when app is open
> >     2. Menu > shutdown
> 
> that option doesn't have any effect
> 
> 

After `Menu > shutdown` the app still runs in the background?

**Comment by amosstevens** [comment]
if the notification is turned off and the app is in background this is an unwanted behavior, someone shouldn't shutdown the application manually

**Comment by nahuhh** [comment]
> if the notification is turned off and the app is in background this is an unwanted behavior, someone shouldn't shutdown the application manually

Are you confirming that 
1. Select "run when app is open"
2. Close app via swiping away from recent 
= app still running?

**Comment by Yobo7000** [comment]
> Are you confirming that
> 
>     1. Select "run when app is open"
> 
>     2. Close app via swiping away from recent
>        = app still running?

I don't have such option "run when app is open".

I have all other apps running as expected. the problem about SimpleX looks like disabling notifications doesn't work at all so it keeps communicating even with that options is set off. another problem is that SimpleX doesn't care when you restrict battery for the app. it just keeps communicating rarely (when you leave traffic listening over night)

I guess SimpleX android app by design is supposed to work as I expect but those features aren't implemented well so this is kinda bug

**Comment by amosstevens** [comment]
if "runs when app is open" is selected and the app still communicate with the server when it's in background, it's an unwanted behavior

i have used pcap with the notification off on simplex, and it doesn't seems to connect to the server

i have simplex 5.8 beta 3
android 13, grapheneOS

**Comment by Yobo7000** [comment]
> i have used pcap with the notification off on simplex, and it doesn't seems to connect to the server

leave it for 24h

**Comment by nahuhh** [comment]
simplex > menu > Setting > notification > notification service > run when app is open

![0hlet44xkvq0m8o2.jpg](https://github.com/simplex-chat/simplex-chat/assets/50635951/35c556f0-bbf7-45d7-b9f6-58f2a70037df)

**Comment by Yobo7000** [comment]
> simplex > menu > Setting > notification > notification service > run when app is open

yes, friend. I got this option enabled and the app still makes connections

**Comment by epoberezkin** [comment]
The confusion here is in the definition of "open" - whenever the app is present in Recents it is "open" technically, same as it can be opened on desktop, even if it is in background and you don't see its UI.

This term is used to mean "opened (possibly in background)" - as opposed to "active and in foreground".

We will reconsider the terminology in the UX, but in any case to make the app "closed" you need to swipe it up from recents - in this case, if this notification mode is selected, the app will not be communicating with the servers.

**Comment by nahuhh** [comment]
text change from `run when app is open` >> `enabled when app is running`

**Comment by Yobo7000** [comment]
> The confusion here is in the definition of "open" - whenever the app is present in Recents it is "open" technically, same as it can be opened on desktop, even if it is in background and you don't see its UI.
> 
> This term is used to mean "opened (possibly in background)" - as opposed to "active and in foreground".
> 
> We will reconsider the terminology in the UX, but in any case to make the app "closed" you need to swipe it up from recents - in this case, if this notification mode is selected, the app will not be communicating with the servers.

on LineageOS 20 when I swipe it up it still communicates, like the app doesn't closes completely even and keeps communicating. I tried all kind of settings and the only way I can stop it from communicating is to set background processes to zero in developer options
